src/common/cockpitpipe.c

```diff
@@ -1505,6 +1529,47 @@ cockpit_pipe_get_stderr (CockpitPipe *self)
   return priv->err_buffer;
 }
 
+gchar *
+cockpit_pipe_take_stderr_as_utf8 (CockpitPipe *self)
+{
+  GByteArray *buffer;
+  GBytes *bytes;
+  GBytes *clean;
+  gchar *data;
+  gsize length;
+
+  buffer = cockpit_pipe_get_stderr (self);
+  if (!buffer)
+    return NULL;
+
+  /* A little more complicated to avoid big copies */
+  g_byte_array_ref (buffer);
+  g_byte_array_append (buffer, (guint8 *)"x", 1); /* place holder for null terminate */
```

Comment:
...you're modifying the internal buffer here

Reply:
I don't think this a problem, or is it?